core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -525,14 +531,14 @@ private[spark] class MesosClusterScheduler(
         d.retryState.get.nextRetry.before(currentTime)
       }
 
-      scheduleTasks(
+      currentOffers = scheduleTasks(
```

Comment:
Actually you're right we're just returning the value as-is, we can remove the assignment.